src/main/java/com/hubspot/jinjava/lib/tag/eager/EagerIfTag.java

```diff
@@ -0,0 +1,130 @@
+package com.hubspot.jinjava.lib.tag.eager;
+
+import com.hubspot.jinjava.interpret.DeferredValueException;
+import com.hubspot.jinjava.interpret.JinjavaInterpreter;
+import com.hubspot.jinjava.interpret.TemplateSyntaxException;
+import com.hubspot.jinjava.lib.tag.ElseIfTag;
+import com.hubspot.jinjava.lib.tag.ElseTag;
+import com.hubspot.jinjava.lib.tag.IfTag;
+import com.hubspot.jinjava.tree.Node;
+import com.hubspot.jinjava.tree.TagNode;
+import com.hubspot.jinjava.util.LengthLimitingStringBuilder;
+import com.hubspot.jinjava.util.ObjectTruthValue;
+import org.apache.commons.lang3.StringUtils;
+
+public class EagerIfTag extends EagerTagDecorator<IfTag> {
+
+  public EagerIfTag() {
+    super(new IfTag());
+  }
+
+  public EagerIfTag(IfTag ifTag) {
+    super(ifTag);
+  }
+
+  @Override
+  public String eagerInterpret(TagNode tagNode, JinjavaInterpreter interpreter) {
+    if (StringUtils.isBlank(tagNode.getHelpers())) {
+      throw new TemplateSyntaxException(
+        interpreter,
+        tagNode.getMaster().getImage(),
+        "Tag 'if' expects expression"
+      );
+    }
+
+    LengthLimitingStringBuilder result = new LengthLimitingStringBuilder(
+      interpreter.getConfig().getMaxOutputSize()
+    );
+
+    result.append(
+      executeInChildContext(
+        eagerInterpreter ->
+          getEagerImage(tagNode.getMaster(), eagerInterpreter) +
+          renderChildren(tagNode, eagerInterpreter),
+        interpreter,
+        false
+      )
+    );
+    tagNode.getMaster().setRightTrimAfterEnd(false);
+    result.append(reconstructEnd(tagNode));
+
+    return result.toString();
+  }
+
+  @Override
+  public String renderChildren(TagNode tagNode, JinjavaInterpreter interpreter) {
+    // If the branch is impossible, it should be removed
+    boolean definitelyDrop = shouldDropBranch(tagNode, interpreter);
+    // If the branch would definitely get executed, change it to and else
```

Comment:
This comment is a little unclear to me

Reply:
Sorry, I have a typo in it. It's saying that if an `elseif` branch would definitely get executed, it can be changed to an `else`.
Ex.
```
{% if deferred %}
1.
{% elseif foo > 0 %}
2.
{% else %}
3.
{% endif %}
```
if `foo == 2`, then it could get output like:
```
{% if deferred %}
1.
{% else %}
2.
{% endif %}
```

Reply:
And the opposite for `definitelyDrop`. If there were to be a tag like: `{% elseif foo == 0 %}`, it could get dropped as it's impossible.